### ISSUE TITLE
How to add the codetabs in markdown file for dokkahtml generating

### ISSUE THREAD

**Issue by alkuuuo** [Kotlin/dokka#2781]
**Question**
I am trying to document the markdown file that will be generated by dokkahtml. And we have a requirement which is adding the code tabs to display different languages. I already tried some markdown syntax to build it. But it didn't work. How can I do for this case? 

**Screenshots**
The example of expected result:
![image](https://user-images.githubusercontent.com/120694767/208005073-ea3e0ade-2990-45f4-ba32-9f8a524a14fa.png)

**Installation**
- Operating system: Windows
- Build tool: Gradle v7.3.3
- Dokka version: 1.7.20

**Comment by IgnatBeresnev** [comment]
TLDR: not possible out of the box (format limitation), but, depending on the case, can be achieved with a custom Dokka plugin.

Markdown itself (as a markup language) does not provide/have the ability to create such tabs out of the box, so it's not generated by Dokka either.

However, depending on how you host/view your Markdown documentation, it might be possible. 

For instance, Material for MkDocs has an extension that adds tabbed content: https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#usage. Alternatively, Markdown allows using the `details` html tag, which might contain a class - this can be used for tabs, but still needs post-processing by the documentation renderer.

However, Dokka doesn't know anything about any special and specific syntax/tags as it tries to be universal.

So if you want to support generating tabs with your special syntax (like for MkDocs or via `<details>`), you need to create a Dokka plugin and override [buildPlatformDependent](https://github.com/Kotlin/dokka/blob/1.7.20/plugins/gfm/src/main/kotlin/org/jetbrains/dokka/gfm/renderer/CommonmarkRenderer.kt#L142) in your renderer. See [Developer Guides](https://kotlin.github.io/dokka/1.7.20/developer_guide/introduction/) for more info.

Let us know if you have any further questions.

**Comment by IgnatBeresnev** [comment]
Looks like the question has been answered :)
